### PR TITLE
docs: improve NuGet documentation

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Use `1.` for ordered list, fix whitespace
- Fix typos/grammar/capitalization
- Rewrite sentences for clarity
- Rewrite sentences to use the active voice

## Context:

Helps with issue #7626.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] Unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/nuget.md [hunk 1]
```diff
@@ -39,15 +39,15 @@ Alternative feeds can be specified either [in a `NuGet.config` file](https://doc
 }
 ```
 
-If this example we defined 3 NuGet feeds.
-Packages resolving will process feeds consequentially.
-It means that if package will be resolved in second feed renovate won't look in last one.
+In this example we defined 3 NuGet feeds.
+Package resolving processes feeds in the listed order, from top to bottom.
+This means that if a package is resolved in the second feed, Renovate won't check the third and final feed.
```
Comment (viceice) [inline docs/usage/nuget.md:44]: Nuget datasource will use `merge` registry strategy, so will search all registries and merge the results

Comment (HonkingGoose) [inline docs/usage/nuget.md:44]:
```suggestion
In this example we defined 3 NuGet feeds.
The package resolving process searches these 3 feeds for package updates and merge the results.
```

How about something like this?
I don't know what happens after the `merge` strategy is done, does the bot open a PR then?
Maybe we should cover that???

Comment (viceice) [reply docs/usage/nuget.md:44]: `merge` means join all resolved versions to a single dependency (removing duplicate versions)

Comment (HonkingGoose) [reply docs/usage/nuget.md:44]:
```suggestion
In this example we defined 3 NuGet feeds.
The package resolving process uses the `merge` strategy to handle the 3 feeds.
All feeds are checked for dependency updates, and duplicate updates are merged/joined together into a single dependency update.
```

Comment (viceice) [reply docs/usage/nuget.md:44]: Sounds good

**Review by rarkins — APPROVED** [review]

File: docs/usage/nuget.md (hunk 1 above)
Comment (HonkingGoose) [reply docs/usage/nuget.md:44]: Opened PR #7841 to get this into the docs.

**Pull request merged**